spec/classes/pulpcore_spec.rb

```diff
@@ -19,6 +19,22 @@
             .that_notifies(['Class[Pulpcore::Database]', 'Class[Pulpcore::Service]'])
         end
       end
+
+      context 'with external postgresql' do
+        let :params do
+          {
+            postgresql_db_host: '192.0.2.222',
+            postgresql_manage_db: false,
+          }
+        end
+
+        it do
+          is_expected.to compile.with_all_deps
+          is_expected.not_to contain_class('postgresql::server::db')
```

Comment:
The DB is a define so this will always be true. I'd recommend adding a check i the default case as `is_expected.to` and here as `not_to`. Probably like this
```suggestion
          is_expected.not_to contain_class('postgresql::server')
          is_expected.not_to contain_postgresql__server__db('pulpcore')
```

Reply:
@ekohl super helpful feedback. Thanks as always.